lib/ansible/modules/cloud/amazon/rds_instance.py

```diff
@@ -0,0 +1,1236 @@
+#!/usr/bin/python
+# Copyright (c) 2018 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: rds_instance
+version_added: "2.7"
+short_description: Manage RDS instances
+description:
+    - Create, modify, and delete RDS instances.
+
+requirements:
+    - botocore
+    - boto3
+extends_documentation_fragment:
+    - aws
+    - ec2
+author:
+    - Sloane Hertel (@s-hertel)
+
+options:
+  # General module options
+    state:
+        description:
+          - Whether the snapshot should exist or not. I(rebooted) is not idempotent and will leave the DB instance in a running state
+            and start it prior to rebooting if it was stopped. I(present) will leave the DB instance in the current running/stopped state,
+            (running if creating the DB instance).
+          - I(state=running) and I(state=started) are synonyms, as are I(state=rebooted) and I(state=restarted). Note - rebooting the instance
+            is not idempotent.
+        choices: ['present', 'absent', 'running', 'started', 'stopped', 'rebooted', 'restarted']
+        default: 'present'
+    creation_source:
+        description: Which source to use if restoring from a template (an existing instance, S3 bucket, or snapshot).
+        choices: ['snapshot', 's3', 'instance']
+    force_update_password:
+        description:
+          - Set to True to update your cluster password with I(master_user_password). Since comparing passwords to determine
+            if it needs to be updated is not possible this is set to False by default to allow idempotence.
+        type: bool
+        default: False
+    promote:
```

Comment:
while you've implemented `promote` in a declarative fashion, the parameter name doesn't seem declarative.

I wonder if there's a better way to do this. In my original attempt at rds_instance, I used the fact a db was a replication_source, when passed without a source_db_identifier, to promote, but that seems far too dangerous in hindsight.

`read_replica: False` might be better, with `read_replica: True` being implied if `source_db_identifier` is passed.

Reply:
That sounds much better, thanks for the suggestion. I will update that.